### ISSUE TITLE
v2.0.0 - Conversion to Typescript/Bridged Camera

### ISSUE THREAD

**Issue by mrMiimo** [homebridge-plugins/homebridge-camera-ffmpeg#527]
### Breaking Changes

* Code has been refactored to typescript, thanks to [Brandawg93](https://github.com/Brandawg93).
* Plugin requires homebridge >= 1.0.0.
* Cameras no longer need to be manually added to homebridge
	* Cameras are now bridged instead of being created as external accessories in homebridge. 
	* Once you update, you will see two copies of each of your cameras.
	* You will need to manually remove the old cameras from HomeKit by going into the cameras' settings and choosing "Remove Camera from Home". 
	* The new bridged cameras will not have this option, and will instead have a "Bridge" button.
	* You will also need to copy over any automations that you had tied to your cameras, such as motion detection.

#### Other Changes

* Google Drive Upload has been removed in this update. PRs are welcome for other Video Cloud Options.

To Test 
```
sudo npm install -g homebridge-camera-ffmpeg@beta
```
To Revert
```
sudo npm install -g homebridge-camera-ffmpeg@latest
```

**Comment by donavanbecker** [comment]
@mrMiimo Thanks for reporting this. We aren't ready to test the beta as of now. We have to make a few more adjustments before it is ready for testing. I will Leave this Issue open though for when we are ready.

**Comment by donavanbecker** [comment]
@mrMiimo `v2.0.1-beta.4` should have the fix. If you would like to test.

`sudo npm install -g --unsafe-perm homebridge-camera-ffmpeg@beta`

**Comment by mrMiimo** [comment]
Tested, it works!  Thanks...
If you need to do more tests, know that I would be glad to help.

**Comment by joffreyha** [comment]
Tested v2.0.1-beta.5 and works fine for me, had to rename the camera for other Apps to pickup the change. Homecam or example had an issue displaying the cam in the widget, but worked fine after a rename.

**Comment by mrMiimo** [comment]
If I had a mqtt server with user and password. Can I set these variables in the camera-ffmpeg configuration file ?

**Comment by donavanbecker** [comment]
> If I had a mqtt server with user and password. Can I set these variables in the camera-ffmpeg configuration file ?

@mrMiimo what variables? MQTT Support is not enabled.

**Comment by mrMiimo** [comment]
> > If I had a mqtt server with user and password. Can I set these variables in the camera-ffmpeg configuration file ?
> 
> @mrMiimo what variables? MQTT Support is not enabled.

upsss sorry, I didn't understand..

**Comment by hassbian-ABC** [comment]
"audio": true， But there was no sound

**Comment by donavanbecker** [comment]
> "audio": true， But there was no sound

Yes this is something that needs to be worked out. Thanks!

**Comment by malhal** [comment]
Hi I can help test this but I'd like to know what "Cameras are now bridged instead of being created as external accessories in homebridge." means, thanks.

**Comment by donavanbecker** [comment]
> Hi I can help test this but I'd like to know what "Cameras are now bridged instead of being created as external accessories in homebridge." means, thanks.

That means that you don’t have to add the cameras separately, once you add the base stations it will add the cameras also.

**Comment by sergoo-lbt** [comment]
The sound still doesn't work

**Comment by donavanbecker** [comment]
@sergoo-lbt Yes. If anyone wants to submit a pr, you are more then welcome.

**Comment by donavanbecker** [comment]
@sergoo-lbt and @hassbian-ABC with `beta18` audio should be working. If you can test?

**Comment by sergoo-lbt** [comment]
> @sergoo-lbt and @hassbian-ABC with `beta18` audio should be working. If you can test?

Everything is fine now!  You are great fellows!

**Comment by sergoo-lbt** [comment]
It looks like the copy codec doesn't copy the camera parameters. In the camera settings, the bitrate is set to 2048 and the homebridge log shows 300 kbit / sec

**Comment by sergoo-lbt** [comment]
@donavanbecker , Absolutely sure. I used to work on this bitrate.

**Comment by donavanbecker** [comment]
> @donavanbecker , Absolutely sure. I used to work on this bitrate.

pull request to fix this is welcomed.

**Comment by donavanbecker** [comment]
released `v2.0.0`, so closing.
